pre_commit_hooks/update_commit_msg.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, unicode_literals
+import sys
+import optparse
+from invoke import run
+
+import re
+
+
+class UpdateCommitMsg(object):
```

Comment:
this is unnecessarily object-oriented.  A better approach (imo) is to have a main function which calls out to the check_file function.  All of the other hooks in this repository are written in that way so you can find a few examples.  My bare minimum setup usually looks like this:

```python
import argparse

...

def main(argv=None):
    parser = argparse.ArgumentParser()
    parser.add_argument('filenames', nargs='+')
    args = parser.parse_args(argv)

    ...

if __name__ == '__main__':
    exit(main())
```

this allows you to have a testable interface (`parser.parse_args(None)` will fall back to sys.argv)

Reply:
Sounds fair.